crates/wasm-smith/src/core.rs

```diff
@@ -1472,25 +1472,28 @@ impl Module {
             Ref,
         }
 
-        match u.arbitrary::<ValTypeClass>()? {
+        let mut val_classes: Vec<_> = self
+            .valtypes
+            .iter()
+            .map(|vt| match vt {
+                ValType::I32 => ValTypeClass::I32,
+                ValType::I64 => ValTypeClass::I64,
+                ValType::F32 => ValTypeClass::F32,
+                ValType::F64 => ValTypeClass::F64,
+                ValType::V128 => ValTypeClass::V128,
+                ValType::Ref(_) => ValTypeClass::Ref,
+            })
+            .collect();
+        val_classes.sort_unstable();
+        val_classes.dedup();
+
+        match u.choose(&val_classes)? {
             ValTypeClass::I32 => Ok(ValType::I32),
             ValTypeClass::I64 => Ok(ValType::I64),
             ValTypeClass::F32 => Ok(ValType::F32),
             ValTypeClass::F64 => Ok(ValType::F64),
-            ValTypeClass::V128 => {
-                if self.config.simd_enabled {
-                    Ok(ValType::V128)
-                } else {
-                    Ok(ValType::I32)
-                }
-            }
-            ValTypeClass::Ref => {
-                if self.config.reference_types_enabled {
-                    Ok(ValType::Ref(self.arbitrary_ref_type(u)?))
-                } else {
-                    Ok(ValType::I32)
-                }
-            }
+            ValTypeClass::V128 => Ok(ValType::V128),
+            ValTypeClass::Ref => Ok(ValType::Ref(self.arbitrary_ref_type(u)?)),
```

Comment:
`reference_types_enabled: false` no longer seems to work in my testing, and this looks related.

Reply:
I believe this is fixed - try the latest commit.

Reply:
I'm still seeing reference types.

I'm running this:
```
wasm-tools smith --config <(echo '{ "reference_types_enabled": false }') -t <(head -c 50 /dev/urandom)
```

Reply:
Debugging the actual `config` passed to the `Module` shows that it still has `reference_types_enabled: true`, and seems to be completely ignoring whatever you pass to `--config`. Not sure why - this is unrelated to my changes and repros on main (@fitzgen).

Reply:
@gtrepta looks like you accidentally ran into https://github.com/bytecodealliance/wasm-tools/pull/1824 where the field name is `reference-types-enabled`, with `-` instead of `_`.

I'll note though that with that I'm occasionally seeing panics in `wasm-tools smith ...` on main (not released yet) so I'll look to track that down.

Reply:
The panic should be fixed in https://github.com/bytecodealliance/wasm-tools/pull/1825